tools/please_pex/pex/pex_test_main.py

```diff
@@ -0,0 +1,59 @@
+"""Customised test runner to output in JUnit-style XML."""
+
+import os
+import sys
+
+# This will get templated in by the build rules.
+TEST_NAMES = '__TEST_NAMES__'.split(',')
+
+
+def initialise_coverage():
+    """Imports & initialises the coverage module."""
+    import coverage
+    from coverage import control as coverage_control
+    _original_xml_file = coverage_control.XmlReporter.xml_file
+    # Fix up paths in coverage output which are absolute; we want paths relative to
+    # the repository root. Also skip empty __init__.py files.
+
+    def _xml_file(self, fr, analysis, *args, **kvargs):
+        if PEX_PATH in fr.filename:
+            fr.filename = fr.filename[len(PEX_PATH) + 1:]  # +1 to remove the remaining /
+        if fr.filename == '__main__.py':
+            return  # Don't calculate coverage for the synthetic entrypoint.
+        if not (fr.filename.endswith('__init__.py') and len(analysis.statements) < 1):
+            analysis.filename = fr.filename
+            fr.relname = fr.filename
+            _original_xml_file(self, fr, analysis, *args, **kvargs)
+    coverage_control.XmlReporter.xml_file = _xml_file
+    return coverage
+
+
+def main():
+    """Runs the tests. Returns an appropriate exit code."""
+    args = [arg for arg in sys.argv[1:]]
+    # Add .bootstrap dir to path, after the initial pex entry
+    sys.path = sys.path[:1] + [os.path.join(sys.path[0], '.bootstrap')] + sys.path[1:]
```

Comment:
I think this isn't needed anymore, given that the `.bootstrap` dir shouldn't exist?